ksql-engine/src/test/java/io/confluent/ksql/structured/SchemaKStreamTest.java

```diff
@@ -412,47 +508,72 @@ public void shouldUseFactoryForGrouped() {
         buildSchemaKStream(ksqlStream, mockKStream, mockGroupedFactory, mockJoinedFactory);
 
     // When:
-    initialSchemaKStream.groupBy(leftSerde, groupByExpressions, GROUP_OP_NAME);
+    initialSchemaKStream.groupBy(
+        leftSerde,
+        groupByExpressions,
+        childContextBuilder);
 
     // Then:
-    verify(mockKStream, mockGroupedFactory);
+    verify(mockGroupedFactory).create(
+        eq(StreamsUtil.buildOpName(childContextBuilder.getQueryContext())),
+        any(StringSerde.class),
+        same(leftSerde));
+    verify(mockKStream).groupBy(any(KeyValueMapper.class), same(grouped));
   }
 
-  private void expectCreateJoined(final String opName, final Serde<GenericRow> rightSerde) {
-    expect(mockJoinedFactory.create(anyObject(StringSerde.class), same(leftSerde), same(rightSerde), eq(opName)))
-        .andReturn(joined);
-    replay(mockJoinedFactory);
+  private void whenCreateJoined() {
```

Comment:
Could simply put this into `init`, since I see five tests that require it and none that overwrite it.